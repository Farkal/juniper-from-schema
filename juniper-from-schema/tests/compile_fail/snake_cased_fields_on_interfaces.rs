#![allow(dead_code, unused_variables, unused_must_use, unused_imports)]
include!("../compile_pass/setup.rs");

juniper_from_schema::graphql_schema! {
    type Query {
        field: SomeInterface!
    }

    schema { query: Query }

    interface SomeInterface {
        snake_cased: String!
    }
}

pub struct Query;

impl QueryFields for Query {
    fn field_field<'a>(
        &self,
        _: &Executor<'a, Context>,
        _: &QueryTrail<'a, SomeInterface, Walked>,
    ) -> FieldResult<&SomeInterface> {
        unimplemented!()
    }
}
