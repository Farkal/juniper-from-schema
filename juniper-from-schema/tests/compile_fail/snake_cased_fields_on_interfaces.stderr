error: proc macro panicked
  --> $DIR/snake_cased_fields_on_interfaces.rs:4:1
   |
4  | / graphql_schema! {
5  | |     type Query {
6  | |         field: SomeInterface!
7  | |     }
...  |
13 | |     }
14 | | }
   | |_^
   |
   = help: message: 
           
           [91merror[0m: Field names must be camelCase, not snake_case
            --> schema:2:17
             |
           2 |    SomeInterface { snake_cased : String ! }
             |                    [91m^[0m
           
           This is because Juniper always converts all field names to camelCase
           
           
           aborting due to previous error
           

error: could not compile `juniper-from-schema-tests`.
